repository/src/main/java/org/apache/atlas/accesscontrol/persona/callable/CreateRangerPolicyWorker.java

```
@@ -0,0 +1,72 @@
+                                    AtlasRangerService atlasRangerService) {
+        this.context = context;
+        this.provisionalPolicy = provisionalPolicy;
+        this.atlasRangerService = atlasRangerService;
+    }
+
+    @Override
+    public RangerPolicy call() {
+        RangerPolicy ret = null;
+        LOG.info("Starting CreateRangerPolicyWorker");
+
+        RangerPolicy rangerPolicy = null;
+        try {
+            //check if there is existing Ranger policy of current provisional Ranger policy
+            rangerPolicy = AccessControlUtil.fetchRangerPolicyByResources(atlasRangerService,
+                    context.isDataPolicy() ? RANGER_HEKA_SERVICE_TYPE.getString() : RANGER_ATLAS_SERVICE_TYPE.getString(),
+                    context.isDataMaskPolicy() ? RANGER_POLICY_TYPE_DATA_MASK : RANGER_POLICY_TYPE_ACCESS,
+                    provisionalPolicy);
+
+            if (rangerPolicy == null) {
+                ret = atlasRangerService.createRangerPolicy(provisionalPolicy);
+            } else {
+                if (context.isDataMaskPolicy()) {
+                    rangerPolicy.getDataMaskPolicyItems().add(provisionalPolicy.getDataMaskPolicyItems().get(0));
+                } else if (context.isAllowPolicy()) {
+                    rangerPolicy.getPolicyItems().add(provisionalPolicy.getPolicyItems().get(0));
+                } else {
+                    rangerPolicy.getDenyPolicyItems().add(provisionalPolicy.getDenyPolicyItems().get(0));
+                }
+
+                List<String> labels = rangerPolicy.getPolicyLabels();
+                labels.addAll(AtlasPersonaUtil.getLabelsForPersonaPolicy(context.getPersona().getGuid(), context.getPersonaPolicy().getGuid()));
+                rangerPolicy.setPolicyLabels(labels);
+
+                ret = atlasRangerService.updateRangerPolicy(rangerPolicy);
+            }
+        } catch (AtlasBaseException e) {
+            LOG.error("Failed to create Ranger policies: {}", e.getMessage());
```

api should throw an error